### PR TITLE
add max restart count limit for failed container

### PR DESCRIPTION
**Pull request by kadisi**
Signed-off-by: kadisi <iamkadisi@163.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
add max restart count limit for failed container
```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 1]
```diff
@@ -58,6 +58,9 @@ const (
 
 	// The expiration time of version cache.
 	versionCacheTTL = 60 * time.Second
+
+	// faild container max restart count
+	maxContainerRestartCount = 12
```
Comment (krmayankk) [inline pkg/kubelet/kuberuntime/kuberuntime_manager.go]: why is this hardcoded to 12 and why do we need to limit this ?

Comment (kadisi) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go]: if not limit restart count, failed container will restart unlimited.  i think it is not necessary .

Comment (kadisi) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go]: ？

**Comment by idealhack** [comment]
@kadisi Thanks for your work.

AFAIK this is by design. Usually we should have an issue before a PR for this.

/sig node

**Comment by idealhack** [comment]
I'll close this for now. If there are any discussions suggesting we need this change, please feel free to link them.

/close

**Pull request closed without merging**